script_1.js

```diff
@@ -0,0 +1,185 @@
+
+  let currentQuestionIndex = 0;
+  let score = [0];
+  let player = "";
+  let currentPlayerIndex = 0;
+
+  startButton.addEventListener("click", startButtonClick);
+
+  function startButtonClick() {
+    const playerName = playerNameInput.value.trim();
+
+    if (playerName) {
+      localStorage.setItem("playerName", playerName);
+      player = playerName;
+      divApp.classList.add("hidden");
+      quizSection.classList.remove("hidden");
+      playerNameElement.textContent = "PLAYER: " + playerName;
+      playerNameElement.style.textTransform = "uppercase";
+      startQuiz(questions);
+    } else {
+      alert("Please enter your name.");
+    }
+  }
+
+  function startQuiz(questions) {
+    currentQuestionIndex = 0;
+    currentPlayerIndex = 0;
+    score = [0];
+    nextButton.innerHTML = "Next";
+    shuffleButton.innerHTML = "Shuffle Answers";
+    shuffleButton.style.display = "block";
+    showQuestion(questions);
+  }
+
+  function showQuestion(questions) {
+    resetState();
+    let currentQuestion = questions[currentQuestionIndex];
+    let questionNumber = currentQuestionIndex + 1;
+    questionElement.innerHTML =
+      questionNumber + ". " + currentQuestion.question;
+
+    currentQuestion.answers.forEach((answer) => {
+      const button = document.createElement("button");
+      button.innerHTML = answer.text;
+      button.classList.add("btn");
+      answerButtons.appendChild(button);
+
+      if (answer.isCorrect) {
+        button.dataset.isCorrect = answer.isCorrect;
+      }
+
+      button.addEventListener("click", selectAnswer);
+    });
+  }
+
+  function resetState() {
+    nextButton.style.display = "none";
+    explanationElement.innerHTML = "";
+    while (answerButtons.firstChild) {
+      answerButtons.removeChild(answerButtons.firstChild);
+    }
+  }
+
+  function selectAnswer(e) {
+    const selectButton = e.target;
+    const isCorrect = selectButton.dataset.isCorrect === "true";
+
+    if (isCorrect) {
+      selectButton.classList.add("isCorrect");
+      score[currentPlayerIndex]++;
+    } else {
+      selectButton.classList.add("isFalse");
+    }
+
+    explanationElement.innerHTML = questions[currentQuestionIndex].explanation;
+    nextButton.style.display = "block";
+  }
+
+  function showScore() {
+    resetState();
+    questionElement.innerHTML =
+      "The score for: " +
+      player +
+      " is : " +
+      score[currentPlayerIndex] +
+      " of " +
+      questions.length;
+
+    nextButton.innerHTML = "Start the quiz again";
+    nextButton.style.display = "block";
+    shuffleButton.style.display = "none";
+  }
+
+  function goToNextButton() {
+    if (currentPlayerIndex === 0) {
+      currentQuestionIndex++;
+    }
+
+    if (currentQuestionIndex < questions.length) {
+      showQuestion(questions);
+    } else {
+      showScore();
+    }
+  }
+
+  function shuffleAnswers() {
+    for (let i = answerButtons.children.length; i >= 0; i--) {
```

Comment:
Usually in the programming world it is conventional to go from 0 to the length rather than the other way.
Of course both ways are correct :)